### PR TITLE
Fix layer 2 fee alignment

### LINKED ISSUE #17426: [Bug]: Optimism fee on Approve ERC20 Confirmation screen displayed with wrong alignment (on the left instead of right)
### Describe the bug

**Problem**: when I am on the Optimism network, if I try to approve an ERC20 token I can see how the fees value is displayed on the left instead of on the right.

Also, not sure how many decimals should we display in this value,as I believe it should be the same value as Gas fee right? @bschorchit @danjm 

image

### Steps to reproduce

1. Change to Optimism network
2. Go to Etherscan to interact with an ERC20 token https://optimistic.etherscan.io/token/0x94b008aa00579c1307b0ef2c499ad98a8ce58e58#writeContract
3. Click Connect MM
4. Accept MM
5. Click Approve and Add spender and amount
6. Click Write
7. Add any amount on MM
8. Click Next
9. See alignment for fees

### Error messages or log output

_No response_

### Version

10.24.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amerkadicE**
## Explanation

This PR fixes alignment for layer 2 gas fee value.
It also sets layer 2 gas fee decimals to be 12 instead of 6.

Fixes #17426 
### Before
image

### After
image

## Manual Testing Steps

1. Change to Optimism network
2. Go to Etherscan to interact with an ERC20 token https://optimistic.etherscan.io/token/0x94b008aa00579c1307b0ef2c499ad98a8ce58e58#writeContract
3. Click Connect MM
4. Accept MM
5. Click Approve and Add spender and amount
6. Click Write
7. Add any amount on MM
8. Click Next
9. See alignment for fees

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by dzfjz** [comment]
Verified by QA

File: test/e2e/tests/custom-token-add-approve.spec.js [hunk 1]
```diff
@@ -294,7 +294,7 @@ describe('Create token, approve token and approve token without gas', function (
 
         await driver.waitForSelector({
           css: '.box--flex-direction-row > h6',
-          text: '0.0006 ETH',
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply test/e2e/tests/custom-token-add-approve.spec.js:297]: Doesn't appear to be a test related to the change

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/components/app/approve-content-card/approve-content-card.js [hunk 2]
```diff
@@ -135,9 +135,12 @@ export default function ApproveContentCard({
...
                       color={TextColor.textMuted}
                       as="h6"
+                      display={DISPLAY.FLEX}
+                      justifyContent={JustifyContent.spaceBetween}
+                      width={BLOCK_SIZES.FULL}
                     >
-                      <span>{t('transactionDetailLayer2GasHeading')}</span>
-                      {`${ethTransactionTotal} ${nativeCurrency}`}
+                      <Box>{t('transactionDetailLayer2GasHeading')}</Box>
+                      <Box>{`${ethTransactionTotal} ${nativeCurrency}`}</Box>
```
[the remark this answers is not in the document]
Comment (amerkadicE) [reply ui/components/app/approve-content-card/approve-content-card.js]: It needed to wrap the second paragraph in a span.

File: ui/helpers/utils/confirm-tx.util.ts [hunk 3]
```diff
@@ -16,8 +16,9 @@ export function getHexGasTotal({ gasLimit = '0x0', gasPrice = '0x0' }): string {
 export function addEth(firstValue: string, ...otherValues: string[]): string {
   return otherValues
     .reduce((numericAcc, ethAmount) => {
-      return numericAcc.add(new Numeric(ethAmount, 10)).round(6);
+      return numericAcc.add(new Numeric(ethAmount, 10)).round(12);
```
[the remark this answers is not in the document]
Comment (amerkadicE) [reply ui/helpers/utils/confirm-tx.util.ts]: Fixed, now we pass an optional parameter to set number of decimals

File: test/e2e/tests/custom-token-add-approve.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (amerkadicE) [reply test/e2e/tests/custom-token-add-approve.spec.js:297]: We decided to change the number of decimals to 12 on the whole token allowance screen.

cc @danjm

**Pull request closed without merging**